### PR TITLE
Realized in-memory CRUD [no tests | no configured Logger]

### PR DESCRIPTION
**Pull request by andrsj**
- [x] CRUD in memory
- [x] Structured project
- [ ] Written tests
- [ ] Configured logger with different layers

### CODE REVIEW

File: pkg/logger/logger.go [hunk 1]
```diff
@@ -39,11 +42,33 @@ func New(level string) *Logger {
...
+		New(zerolog.ConsoleWriter{
+			Out:        os.Stdout,
+			TimeFormat: time.RFC3339,
+			FormatLevel: func(i interface{}) string {
+				return strings.ToUpper(fmt.Sprintf("[%s]", i))
+			},
+			FormatMessage: func(i interface{}) string {
+				return fmt.Sprintf("| %s", i)
+			},
+			// FormatCaller: func(i interface{}) string {
```
Comment (andrsj) [inline pkg/logger/logger.go:55]: Leaved it for future exploring

File: pkg/logger/logger.go [hunk 2]
```diff
@@ -39,11 +42,33 @@ func New(level string) *Logger {
...
+			// 	return filepath.Base(fmt.Sprintf("%s", i))
+			// },
+		}).
 		With().
 		Timestamp().
+		// Caller(). // ?
```
Comment (andrsj) [inline pkg/logger/logger.go:61]: `zerolog` is strange for me, but I try to understand what is this)

File: example.http [hunk 3]
```diff
@@ -0,0 +1,64 @@
+@hostname = http://localhost
```
Comment (andrsj) [inline example.http]: I used REST API client which can run requests in .http file for manual testing)

File: internal/app/app.go [hunk 4]
```diff
@@ -18,22 +19,36 @@ func New() (*App, error) {
 	var err error
 	a := &App{}
 
+	// Setup logger
```
Comment (andrsj) [inline internal/app/app.go:22]: I steal the logger idea from `clean template` [I don't know which is the idiomatic way to configure logger [especially for 3th party libraries]]

File: internal/app/service/anime.go [hunk 5]
```diff
@@ -1,19 +1,74 @@
 package service
 
-import "github.com/andrsj/go_anime_crud/pkg/logger"
+import (
+	"github.com/andrsj/go_anime_crud/internal/app/model"
+	"github.com/andrsj/go_anime_crud/internal/app/repository"
+	"github.com/andrsj/go_anime_crud/pkg/logger"
+)
 
+// Warning: This interface is partially duplicated from repository
```
Comment (andrsj) [inline internal/app/service/anime.go:9]: Is it good to duplicate methods from another interface?

Comment (andrsj) [reply internal/app/service/anime.go:9]: Ohh :( 

We can do `Embedding interfaces` - my bad... Will fix it later)

File: internal/app/service/anime.go [hunk 6]
```diff
@@ -1,19 +1,74 @@
...
+	return &Service{
+			logger:     l,
+			repository: r,
+		},
+		nil
 }
 
-func (s *Service) Data() string {
-	return "Default"
+func (s *Service) Ping() string {
```
Comment (andrsj) [inline internal/app/service/anime.go]: A dumpling method just for fast checking connection

**Comment by alpaka-alpachino** [comment]
I would suggest to reorganize project structure a bit
- what is the point using `cmd/server/main.go`? Try to avoid complication of the structure without obvious cause - you can use just `cmd/main.go`. Same with `pkg/logger/logger.go` -> `logger/logger.go`. Also `internal/app/service/anime.go` -> `internal/service/anime.go`
- naming is also a very important thing. For example `http` directory name can be changed to `rest` directory to point the design principle that you are using. And as far as you are using RESTful API lets rename also `endpoint` -> `api` and `crud_anime.go` -> `api.go`  (by conventions in go it is a good practice to call files and directories by one word)
- try to remove `app` just use necessary code in main

**Pull request merged**